sycl/test-e2e/Matrix/get_coordinate_ops_impl.hpp

```diff
@@ -0,0 +1,230 @@
+     sycl::accessor acc{buf, cgh, sycl::read_write};
+     sycl::accessor v_rows{sum_rows_v, cgh, sycl::read_write};
+     sycl::accessor v_cols{sum_cols_v, cgh, sycl::read_write};
+
+     cgh.parallel_for<big_matrix<T, NUM_ROWS, NUM_COLS>>(
+         nd_range<2>({NUM_ROWS / SROWS, NUM_COLS / SCOLS * sg_size},
+                     {1, 1 * sg_size}),
+         [=](nd_item<2> spmd_item)
+#ifdef SG_SZ
+             [[intel::reqd_sub_group_size(SG_SZ)]]
+#endif
+         {
+           // The submatrix API has to be accessed by all the workitems in a
+           // subgroup these functions will be called once by the subgroup no
+           // code divergence between the workitems
+           const auto global_idx = spmd_item.get_global_id(0);
+           const auto global_idy = spmd_item.get_global_id(1);
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           sub_group sg = spmd_item.get_sub_group();
+           joint_matrix<sub_group, T, Use, SROWS, SCOLS, Layout> sub;
+           joint_matrix_load(
+               sg, sub, acc.template get_multi_ptr<access::decorated::no>() +
+                   (sg_startx * (SROWS / VF) * NUM_COLS * VF) +
+                   sg_starty / sg_size * SCOLS * VF,
+               NUM_COLS * VF);
+
+           TResult sum_local_rows[NUM_ROWS] = {0};
+           TResult sum_local_cols[NUM_COLS] = {0};
+
+           ext::intel::experimental::matrix::joint_matrix_apply(
+               sg, sub, [&](T &x, size_t row, size_t col) {
+                 sum_local_rows[row + global_idx * SROWS] += x;
+                 sum_local_cols[col + global_idy / sg_size * SCOLS] += x;
+               });
+
+           for (int i = 0; i < NUM_ROWS; i++) {
+             sum_local_rows[i] =
+                 reduce_over_group(sg, sum_local_rows[i], sycl::plus<>());
+             // only Groups leader perform the global reduction
+            if (global_idy % sg_size == 0) {
+                sycl::atomic_ref<TResult, sycl::memory_order::relaxed,
+                                sycl::memory_scope::device>
+                    aref(v_rows[i]);
+                aref.fetch_add(sum_local_rows[i]);
+            }
+           }
+
+           for (int i = 0; i < NUM_COLS; i++) {
+             sum_local_cols[i] =
+                 reduce_over_group(sg, sum_local_cols[i], sycl::plus<>());
+             // only Groups leader perform the global reduction
+             if (global_idy % sg_size == 0) {
+               sycl::atomic_ref<TResult, sycl::memory_order::relaxed,
+                                sycl::memory_scope::device>
+                   aref(v_cols[i]);
+               aref.fetch_add(sum_local_cols[i]);
+             }
+           }
+         }); // parallel for
+   }).wait();
+}
+
+template <typename T, typename TResult, size_t SROWS, size_t SCOLS, use Use, 
+        layout Layout, size_t VF>
+void test_gcop() {
```

Comment:
what does "gcop" stand for?
It is better to spell it out.

Reply:
Thanks for the suggestion! I have changed it.